project1/client/client_tasks/terminate_task.h

```diff
@@ -0,0 +1,37 @@
+#ifndef PROJECT1_TERMINATE_TASK_H
+#define PROJECT1_TERMINATE_TASK_H
+
+#include "../../thread_pool/task.h"
+#include "../../wire_protocol/wire_protocol.h"
+#include "ftp_messages.pb.h"
+#include "../client_util.h"
+
+#include <sys/socket.h>
+#include <utility>
+#include <unistd.h>
+
+namespace client_tasks {
+class TerminateTask : public thread_pool::Task {
```

Comment:
Broadly, I question whether a separate thread is necessary for this. When a terminate request is sent, it seems like all it has to do is connect to a socket and send a short message to the server, which should be quite fast and can therefore probably be done in the main thread.

Reply:
I just made it a separate thread out of convenience. It was honestly a personal preference to separate what happens on the main thread and nport from tport requests.

Should we put it into the main thread?

Reply:
Umm, it's not a huge deal at this point. My very general suggestion is to avoid concurrency whenever possible because it tends to lead to difficult bugs.